Creating OCP performance guide (#564)
RedHatInsights/red-hat-ansible-automation-platform-documentation: downstream/modules/platform/ref-resource-types.adoc

.Example CPU and memory specification
The following cluster has enough free resources to schedule a task pod with a dedicated 100m CPU and 250Mi. 
The cluster can also withstand bursts over that dedicated usage up to 2000m CPU and 2Gi memory.  

[options="nowrap" subs="+quotes,attributes"]
----
spec:
  task_resource_requirements:
    requests:
      cpu: 100m
      memory: 250Mi
    limits:
      cpu: 2000m
      memory: 2Gi
----

{ControllerNameStart} will not schedule jobs that use more resources than the limit set. 
If the task pod does use more resources than the limit set, the container is OOMKilled by kubernetes and restarted.  

== Size recommendations for resource requests

All jobs that use a container group use the same Pod Specification. 
The Pod Specification includes the resource requests for the pod that runs the job. 

All jobs use the same resource requests. 
The specified resource requests for your particular job on the pod specification affect how Kubernetes schedules the job pod based on resources available on worker nodes. 
These are the default values. 

> * In general, 1 fork takes 100Mb of memory. This is set using `system_task_forks_mem`. 

Reviewer: ```suggestion
* One fork typically requires 100Mb of memory. This is set using `system_task_forks_mem`. 
```

> So if your jobs have 5 forks, you would want your job pod specification to request 500Mb of memory.

Reviewer: ```suggestion
If your jobs have 5 forks, the job pod specification must request 500Mb of memory.
```

> * If you have job templates that have particularly high forks value or otherwise need larger resource requests, you should create a separate container group with a different pod spec that indicates larger resource requests. 

Reviewer: ```suggestion
*  For job templates that have a particularly high forks value or otherwise need larger resource requests, create a separate container group with a different pod spec that indicates larger resource requests. 
```

> Then you can assign it to the job template in question. 

Reviewer: ```suggestion
Then you can assign it to the job template. 
```

> For example a job template with the forks value of 50 would be best paired with a container group that requests 5GB of memory. 

Reviewer: ```suggestion
For example a job template with the forks value of 50 must be paired with a container group that requests 5GB of memory. 
```

"Must" might be a bit strong here, but it's probably best to make the forks/pod-spec relationship sound like a requirement

> * If a job’s forks value is very high, such that no single pod would be able to contain the job, use of the job slicing feature can split the inventory up such that the individual job “slices” each do fit well in an automation pod provisioned by the container group.

Reviewer: ```suggestion
* If a the fork value for a job is high enough that no single pod would be able to contain the job, use the job slicing feature.
This splits the inventory up so that individual job “slices” fit in an automation pod provisioned by the container group.
```
